app/components/search-results/row.js

```diff
@@ -0,0 +1,44 @@
+import Component from '@ember/component';
+import { get, computed } from '@ember/object';
+import { alias, or } from '@ember/object/computed';
+
+export default Component.extend({
+  kind: alias('item.kind'),
+  imageUrl: or('item.avatar', 'item.posterImage'),
+  title: or('item.name', 'item.canonicalTitle'),
+  slug: or('item.slug', 'item.id'),
+
+  imageClass: computed('item.kind', function () {
```

Comment:
I don't think `item.kind` is likely to ever change client-side so we shouldn't need to depend on it.

I think it's fine to just use a blank computed property here, another alternative is just to apply this to a property under the `init` func.

Reply:
Assuming that Ember won't ever reuse the same component instance for a different `item` seems like depending on an implementation detail of the view engine, no?